### PR TITLE
Feat/svg icon support

### PR DESCRIPTION
**Pull request by Memoyr**
We're going to need this!

### Benefits

Icons font aren't easy to style via CSS - svg sprite symbol enable that.
https://css-tricks.com/svg-sprites-use-better-icon-fonts/

### CODE REVIEW

**Review by FredericEspiau — DISMISSED** [review]

**Comment by jpatovh** [comment]
Hello @Memoyr , thank you for your PR.
I'll review it; I need to test it before approve it.

File: README.md [hunk 1]
```diff
@@ -17,10 +17,13 @@ This stack core is to be included in your main project and sets up many Gulp tas
 - SCSS => CSS compiling with LibSass, PostCSS, linting, CSScomb(x), and SourceMaps
 - JS compiling via Babel, linting and aggregation
 - webpack module bundling
-- SVG => Font Icons compiling with support for adding mixins and classes to SCSS along with a demo page
 - Drupal file watching to trigger Drush cache clears
 - Copy any files to an other location
 - Sprite generator (with Retina Display support)
+- SVG Icon Support ( generation of `sprite.svg` file as a collection of symbol wich are expected to be implemented with `<use>` https://developer.mozilla.org/en-US/docs/Web/SVG/Element/use )
```
Comment (antleblanc) [inline README.md]: `( generation ` => `(generation`

File: docs/features/spritesvg.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
+It uses [gulp.svg-sprite](https://github.com/jkphl/gulp-svg-sprite) and [gulp.svgmin](https://github.com/ben-eb/gulp-svgmin)
+
```
Comment (antleblanc) [inline docs/features/spritesvg.md]: Could you please remove extra empty lines?

File: docs/features/spritesvg.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+## Usage
+
+- Add your raw svg icon in `config.spritesvg.src`  
+- run `gulp spritesvg`  
+- A  `sprite.svg` file will be generated as a collection of symbol
```
Comment (antleblanc) [inline docs/features/spritesvg.md]: Extra space.

File: docs/features/spritesvg.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+- run `gulp spritesvg`  
+- A  `sprite.svg` file will be generated as a collection of symbol  
+- Implement it with `<use>` [Example](https://developer.mozilla.org/en-US/docs/Web/SVG/Element/use)  
+ - Drupal 8 and twig implementation[ Example](https://www.lullabot.com/articles/better-svg-sprite-reuse-in-drupal-8)  
+ - [check if you need svg4everybody](https://github.com/jonathantneal/svg4everybody#implementation-status)
+
+Notes:
+- symbol IDs are prefixed with `icon-` and append file name  
+- get your designer to make icons monochrome and convert the strokes into paths, also simplify the paths down to single shapes.[Example](https://medium.com/sketch-app-sources/preparing-and-exporting-svg-icons-in-sketch-1a3d65b239bb)  
+
```
Comment (antleblanc) [inline docs/features/spritesvg.md]: Could you please remove extra empty lines?

File: package-lock.json [hunk 5]
```diff
@@ -0,0 +1,16946 @@
+{
```
Comment (antleblanc) [inline package-lock.json]: No more needed since #18.

**Review by antleblanc — COMMENTED** [review]
Nice job @Memoyr!

File: lib/spritesvg.js [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+const path = require('path');
+const glob = require('glob');
+const del = require('del');
+
+module.exports = (gulp, config, tasks) => {
+  function createSpriteSvg() {
+    return glob(config.spritesvg.src, (err, dirs) => {
+      dirs.forEach((dir) => {
+        gulp.src(path.join(dir, '*.svg'))
+          .pipe(rename({ prefix: 'icon-' }))
```
Comment (jpatovh) [inline lib/spritesvg.js]: il faut pouvoir changer ca dans la config

Comment (Memoyr) [reply lib/spritesvg.js]: ok

Comment (Memoyr) [reply lib/spritesvg.js]: Il y aura d'autres améliorations possible à faire avec svg-sprite . Ce pourra être fait ultérieurement.
Pour le prefix, je fais cela tout de suite.

File: package-lock.json (hunk 5 above)
Comment (Memoyr) [reply package-lock.json]: hey @antleblanc , I've tried `git checkout master package-lock.json` and i'ts not doing the trick.
I don't want create a new branch& PR just for that... Do you know how I can solve that ?

File: docs/features/spritesvg.md (hunk 4 above)
Comment (Memoyr) [reply docs/features/spritesvg.md]: done

**Comment by Memoyr** [comment]
Hey @blaryjp ! It would be great if this can be merged soon. I'm already working on the second phase of that feature, which is extending support for multiple directory of images and clean/optimization(svgo)  specifically  according to icon type. As of now, this version will work only for utility icons (close, arrow, magnifying glass, etc.) and since some of our projects need support for simple illustration icon type I'm currently extending the feature.
Merci!

**Comment by jleveugle** [comment]
Hello @Memoyr,
Your pull request is now closed due to inactivity.
Feel free to re-open it if you have any update.
Anna Bot for OVH UX <3

**Comment by jleveugle** [comment]
Hello @Memoyr,
Your pull request is now closed due to inactivity.
Feel free to re-open it if you have any update.
Anna Bot for OVH UX <3

**Comment by jleveugle** [comment]
Hello @Memoyr,
Your pull request is now closed due to inactivity.
Feel free to re-open it if you have any update.
Anna Bot for OVH UX <3

**Pull request closed without merging**